File: backend/app/core/permissions.py
from app.models.user_role import UserRole


PERMISSIONS: list[tuple[str, str]] = [
  ('page.clients.view', 'Видеть страницу Клиенты'),
  ('page.team.view', 'Видеть страницу Команда'),
  ('page.dashboard.view', 'Видеть страницу Дашборд'),
  ('page.projects.view', 'Видеть страницу Проекты'),
  ('page.price-monitoring.view', 'Видеть страницу Ценовой мониторинг'),
  ('page.survey-builder.view', 'Видеть страницу Конструктор анкет'),
  ('page.reports.view', 'Видеть страницу Отчет по анкетам'),
  ('page.permissions.view', 'Видеть страницу Разрешения'),
  ('media.audio.view', 'Видеть аудио'),
  ('media.photo.view', 'Видеть фото'),
  ('media.video.view', 'Видеть видео'),
  ('checks.create', 'Создавать проверки'),
  ('checks.appeal', 'Апеллировать'),
  ('notifications.enabled', 'Получать уведомления'),
  ('checks.edit', 'Редактировать проверки'),
  ('users.edit', 'Редактировать профили пользователей'),
  ('notifications.send', 'Отправлять уведомления'),
  ('notifications.inbox', 'Видеть уведомления'),
  ('data.export', 'Экспорт'),
  ('data.import', 'Импорт'),
  ('tp.fio.view', 'Видеть ФИО ТП'),
  ('survey.pdf.view', 'Видеть PDF анкеты'),
  ('checks.price.view', 'Видеть цену проверки'),
]

PERMISSION_KEYS = {k for k, _ in PERMISSIONS}

BASE_DEFAULT_PERMISSIONS_BY_ROLE: dict[UserRole, list[str]] = {
  UserRole.admin: [k for k, _ in PERMISSIONS],
  UserRole.manager: [k for k, _ in PERMISSIONS if k != 'page.permissions.view'],
  UserRole.controller: [
    'page.dashboard.view',
    'page.projects.view',
    'page.price-monitoring.view',
    'page.reports.view',
    'media.audio.view',
    'media.photo.view',
    'media.video.view',
    'checks.appeal',
    'notifications.enabled',
    'notifications.inbox',
    'checks.edit',
    'data.export',
    'survey.pdf.view',
    'checks.price.view',
  ],
  UserRole.coordinator: [
    'page.team.view',
    'page.dashboard.view',
    'page.projects.view',
    'page.price-monitoring.view',
    'page.survey-builder.view',
    'page.reports.view',
    'media.audio.view',
    'media.photo.view',
    'media.video.view',
    'checks.appeal',
    'notifications.enabled',
    'notifications.inbox',
    'checks.edit',
    'data.export',
    'survey.pdf.view',
    'checks.price.view',
  ],
  UserRole.client: [
    'page.dashboard.view',
    'page.reports.view',
    'media.photo.view',
    'survey.pdf.view',
    'notifications.inbox',
  ],
}

PAGE_PERMISSION_KEYS = {k for k in PERMISSION_KEYS if k.startswith('page.')}

File: src/entities/user/model/permissions.ts
import type { UserRole } from './roles'

export const userPermissions: Array<{ key: string; label: string }> = [
  { key: 'page.clients.view', label: 'Видеть страницу Клиенты' },
  { key: 'page.team.view', label: 'Видеть страницу Команда' },
  { key: 'page.dashboard.view', label: 'Видеть страницу Дашборд' },
  { key: 'page.projects.view', label: 'Видеть страницу Проекты' },
  { key: 'page.price-monitoring.view', label: 'Видеть страницу Ценовой мониторинг' },
  { key: 'page.survey-builder.view', label: 'Видеть страницу Конструктор анкет' },
  { key: 'page.reports.view', label: 'Видеть страницу Отчет по анкетам' },
  { key: 'page.permissions.view', label: 'Видеть страницу Разрешения' },
  { key: 'media.audio.view', label: 'Видеть аудио' },
  { key: 'media.photo.view', label: 'Видеть фото' },
  { key: 'media.video.view', label: 'Видеть видео' },
  { key: 'checks.create', label: 'Создавать проверки' },
  { key: 'checks.appeal', label: 'Апеллировать' },
  { key: 'notifications.enabled', label: 'Получать уведомления' },
  { key: 'checks.edit', label: 'Редактировать проверки' },
  { key: 'users.edit', label: 'Редактировать профили пользователей' },
  { key: 'notifications.send', label: 'Отправлять уведомления' },
  { key: 'notifications.inbox', label: 'Видеть уведомления' },
  { key: 'data.export', label: 'Экспорт' },
  { key: 'data.import', label: 'Импорт' },
  { key: 'tp.fio.view', label: 'Видеть ФИО ТП' },
  { key: 'survey.pdf.view', label: 'Видеть PDF анкеты' },
  { key: 'checks.price.view', label: 'Видеть цену проверки' },
]

const permissionLabelsMap = new Map(userPermissions.map((item) => [item.key, item.label]))

export function userPermissionLabel (key: string) {
  return permissionLabelsMap.get(key) ?? key
}

export const userDefaultPermissionsByRole: Record<UserRole, string[]> = {
  admin: userPermissions.map((x) => x.key),
  manager: userPermissions.filter((x) => x.key !== 'page.permissions.view').map((x) => x.key),
  controller: [
    'page.dashboard.view',
    'page.projects.view',
    'page.price-monitoring.view',
    'page.reports.view',
    'media.audio.view',
    'media.photo.view',
    'media.video.view',
    'checks.appeal',
    'notifications.enabled',
    'notifications.inbox',
    'checks.edit',
    'data.export',
    'survey.pdf.view',
    'checks.price.view',
  ],
  coordinator: [
    'page.team.view',
    'page.dashboard.view',
    'page.projects.view',
    'page.price-monitoring.view',
    'page.survey-builder.view',
    'page.reports.view',
    'media.audio.view',
    'media.photo.view',
    'media.video.view',
    'checks.appeal',
    'notifications.enabled',
    'notifications.inbox',
    'checks.edit',
    'data.export',
    'survey.pdf.view',
    'checks.price.view',
  ],
  client: ['page.dashboard.view', 'page.reports.view', 'media.photo.view', 'survey.pdf.view', 'notifications.inbox'],
}
